packages/x-date-pickers/src/CalendarPicker/PickersCalendarHeader.tsx

```diff
@@ -148,11 +148,11 @@ export function PickersCalendarHeader<TDate>(props: PickersCalendarHeaderProps<T
   const selectPreviousMonth = () => onMonthChange(utils.getPreviousMonth(month), 'right');
 
   const isNextMonthDisabled = useNextMonthDisabled(month, {
-    disableFuture: disableFuture || disabled,
```

Comment:
`disableFuture` is disabling compare to the current day, not the selected one, so it does not make sense, especially for the range picker. I moved the responsibility to adapt `minDate`/`maxDate` when picker is disabled to the parent